File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] suff = new int[n];
        suff[n-1] = 1;
        for(int i = n-2;i>=0;i--){
            suff[i] = suff[i+1]*nums[i+1];
        }
        int pre = 1;
        int[] ans = new int[n];
        for(int i = 0;i<n;i++){
            ans[i] = pre*suff[i];
            if(i != n-1) pre *= nums[i]; 
        }
        return ans;
    }
}  

File: Data Structures & Algorithms/permutations/submission-1.cpp
class Solution {
public:
    void solve(vector<int> &nums,int idx,vector<int> &temp,vector<vector<int>> &ans){
        if(idx >= nums.size()){
            ans.push_back(temp);
            return ;
        }

        for(int i = 0;i<nums.size();i++){
            if(temp[i] == -11){
                temp[i] = nums[idx];
                solve(nums,idx+1,temp,ans);
                temp[i] = -11;
            }
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> ans;
        int n = nums.size();
        vector<int> temp(n,-11);
        solve(nums,0,temp,ans);
        return ans;
    }
};
